Index: actions.class.php
===================================================================
--- actions.class.php	(revision 42)
+++ actions.class.php	(working copy)
@@ -27,6 +27,9 @@
 
     $this->processFilters();
 
+<?php if ($this->getParameterValue('list.filters')): ?>
+    $this->filters = $this->getUser()->getAttributeHolder()->getAll('sf_admin/<?php echo $this->getSingularName() ?>/filters');
+<?php endif ?>
     // pager
     $this->pager = new sfPropelPager('<?php echo $this->getClassName() ?>', <?php echo $this->getParameterValue('list.max_per_page', 20) ?>);
     $c = new Criteria();
@@ -37,9 +40,15 @@
     $this->pager->init();
   }
 
+  public function executeShow ()
+  {
+    $this-><?php echo $this->getSingularName() ?> = <?php echo $this->getClassName() ?>Peer::retrieveByPk(<?php echo $this->getRetrieveByPkParamsForAction(49) ?>);
+    $this->forward404Unless($this-><?php echo $this->getSingularName() ?>);
+  }
+
   public function executeCreate ()
   {
-    return $this->forward('<?php echo $this->getModuleName() ?>', 'edit');
+    $this-><?php echo $this->getSingularName() ?> = new <?php echo $this->getClassName() ?>();
   }
 
   public function executeSave ()
@@ -49,15 +58,12 @@
 
   public function executeEdit ()
   {
-    $this-><?php echo $this->getSingularName() ?> = $this->get<?php echo $this->getClassName() ?>OrCreate();
-
     if ($this->getRequest()->getMethod() == sfRequest::POST)
     {
-      $this-><?php echo $this->getSingularName() ?> = $this->get<?php echo $this->getClassName() ?>OrCreate();
+      $this->save = true;
+      $this->get<?php echo $this->getClassName() ?>OrCreate();
 
-      $this->update<?php echo $this->getClassName() ?>FromRequest();
       $this-><?php echo $this->getSingularName() ?>->save();
-
       $this->setFlash('notice', 'Your modifications have been saved');
 
       if ($this->getRequestParameter('save_and_add'))
@@ -72,6 +78,7 @@
     }
     else
     {
+      $this->get<?php echo $this->getClassName() ?>OrCreate();
       // add javascripts
       $this->getResponse()->addJavascript('/sf/js/prototype/prototype');
       $this->getResponse()->addJavascript('/sf/js/sf_admin/collapse');
@@ -85,8 +92,7 @@
 
     $this-><?php echo $this->getSingularName() ?>->delete();
 
-<?php foreach ($this->getColumnCategories('edit.display') as $category): ?>
-<?php foreach ($this->getColumns('edit.display', $category) as $name => $column): ?>
+<?php foreach ($this->getColumns('') as $name => $column): ?>
 <?php $input_type = $this->getParameterValue('edit.fields.'.$column->getName().'.type') ?>
 <?php if ($input_type == 'admin_input_upload_tag'): ?>
 <?php $upload_dir = $this->getParameterValue('edit.fields.'.$column->getName().'.upload_dir') ?>
@@ -96,37 +102,50 @@
         unlink($currentFile);
       }
 
-<?php endif ?>
-<?php endforeach ?>
-<?php endforeach ?>
+<?php endif; ?>
+<?php endforeach; ?>
+
+<?php foreach ($this->getColumns('') as $name => $column): ?>
+<?php $input_type = $this->getParameterValue('create.fields.'.$column->getName().'.type') ?>
+<?php if ($input_type == 'admin_input_upload_tag'): ?>
+<?php $upload_dir = $this->getParameterValue('create.fields.'.$column->getName().'.upload_dir') ?>
+      $currentFile = sfConfig::get('sf_upload_dir').'/<?php echo $upload_dir ?>/'.$this-><?php echo $this->getSingularName() ?>->get<?php echo $column->getPhpName() ?>();
+      if (is_file($currentFile))
+      {
+        unlink($currentFile);
+      }
+
+<?php endif; ?>
+<?php endforeach; ?>
     return $this->redirect('<?php echo $this->getModuleName() ?>/list');
   }
 
   public function handleErrorEdit()
   {
     $this->preExecute();
-    $this-><?php echo $this->getSingularName() ?> = $this->get<?php echo $this->getClassName() ?>OrCreate();
-    $this->update<?php echo $this->getClassName() ?>FromRequest();
+    $this->save = true;
+    $this->get<?php echo $this->getClassName() ?>OrCreate();
+    //$this->update<?php echo $this->getClassName() ?>FromRequest();
 
     return sfView::SUCCESS;
   }
 
-  protected function update<?php echo $this->getClassName() ?>FromRequest()
+<?php $views = array('edit','create'); foreach ($views as $view): ?>
+  protected function update<?php echo $this->getClassName() ?>From<?php echo ucfirst($view) ?>Request()
   {
     $<?php echo $this->getSingularName() ?> = $this->getRequestParameter('<?php echo $this->getSingularName() ?>');
 
-<?php foreach ($this->getColumnCategories('edit.display') as $category): ?>
-<?php foreach ($this->getColumns('edit.display', $category) as $name => $column): $type = $column->getCreoleType(); ?>
+<?php foreach ($this->getColumns('') as $name => $column): $type = $column->getCreoleType(); ?>
 <?php $name = $column->getName() ?>
 <?php if ($column->isPrimaryKey()) continue ?>
-<?php $credentials = $this->getParameterValue('edit.fields.'.$column->getName().'.credentials') ?>
-<?php $input_type = $this->getParameterValue('edit.fields.'.$column->getName().'.type') ?>
+<?php $credentials = $this->getParameterValue($view.'.fields.'.$column->getName().'.credentials') ?>
+<?php $input_type = $this->getParameterValue($view.'.fields.'.$column->getName().'.type') ?>
 <?php if ($credentials): $credentials = str_replace("\n", ' ', var_export($credentials, true)) ?>
     if ($this->getUser()->hasCredential(<?php echo $credentials ?>))
     {
-<?php endif ?>
+<?php endif; ?>
 <?php if ($input_type == 'admin_input_upload_tag'): ?>
-<?php $upload_dir = $this->getParameterValue('edit.fields.'.$column->getName().'.upload_dir') ?>
+<?php $upload_dir = $this->getParameterValue($view.'.fields.'.$column->getName().'.upload_dir') ?>
     $currentFile = sfConfig::get('sf_upload_dir').'/<?php echo $upload_dir ?>/'.$this-><?php echo $this->getSingularName() ?>->get<?php echo $column->getPhpName() ?>();
     if (!$this->getRequest()->hasErrors() && isset($<?php echo $this->getSingularName() ?>['<?php echo $name ?>_remove']))
     {
@@ -139,10 +158,10 @@
 
     if (!$this->getRequest()->hasErrors() && $this->getRequest()->getFileSize('<?php echo $this->getSingularName() ?>[<?php echo $name ?>]'))
     {
-<?php else: ?>
+<?php elseif ($type != CreoleTypes::BOOLEAN): ?>
     if (isset($<?php echo $this->getSingularName() ?>['<?php echo $name ?>']))
     {
-<?php endif ?>
+<?php endif; ?>
 <?php if ($input_type == 'admin_input_upload_tag'): ?>
       $fileName = md5($this->getRequest()->getFileName('<?php echo $this->getSingularName() ?>[<?php echo $name ?>]').time());
       $ext = $this->getRequest()->getFileExtension('<?php echo $this->getSingularName() ?>[<?php echo $name ?>]');
@@ -163,32 +182,44 @@
         $this-><?php echo $this->getSingularName() ?>->set<?php echo $column->getPhpName() ?>(null);
       }
 <?php elseif ($type == CreoleTypes::BOOLEAN): ?>
-      $this-><?php echo $this->getSingularName() ?>->set<?php echo $column->getPhpName() ?>(isset($<?php echo $this->getSingularName() ?>['<?php echo $name ?>']) ? $<?php echo $this->getSingularName() ?>['<?php echo $name ?>'] : 0);
+    $this-><?php echo $this->getSingularName() ?>->set<?php echo $column->getPhpName() ?>(isset($<?php echo $this->getSingularName() ?>['<?php echo $name ?>']) ? $<?php echo $this->getSingularName() ?>['<?php echo $name ?>'] : 0);
 <?php else: ?>
-      $this-><?php echo $this->getSingularName() ?>->set<?php echo $column->getPhpName() ?>($<?php echo $this->getSingularName() ?>['<?php echo $name ?>']);
-<?php endif ?>
+    $this-><?php echo $this->getSingularName() ?>->set<?php echo $column->getPhpName() ?>($<?php echo $this->getSingularName() ?>['<?php echo $name ?>']);
+<?php endif; ?>
+<?php if ($type != CreoleTypes::BOOLEAN): ?>
     }
+<?php endif; ?>
 <?php if ($credentials): ?>
       }
-<?php endif ?>
-<?php endforeach ?>
-<?php endforeach ?>
+<?php endif; ?>
+<?php endforeach; ?>
   }
 
+<?php endforeach; ?>
   protected function get<?php echo $this->getClassName() ?>OrCreate (<?php echo $this->getMethodParamsForGetOrCreate() ?>)
   {
     if (<?php echo $this->getTestPksForGetOrCreate() ?>)
     {
-      $<?php echo $this->getSingularName() ?> = new <?php echo $this->getClassName() ?>();
+      $this-><?php echo $this->getSingularName() ?> = new <?php echo $this->getClassName() ?>();
+
+      if($this->save)
+      {
+        $this->update<?php echo $this->getClassName() ?>FromCreateRequest();
+      }
     }
     else
     {
-      $<?php echo $this->getSingularName() ?> = <?php echo $this->getClassName() ?>Peer::retrieveByPk(<?php echo $this->getRetrieveByPkParamsForGetOrCreate() ?>);
+      $this-><?php echo $this->getSingularName() ?> = <?php echo $this->getClassName() ?>Peer::retrieveByPk(<?php echo $this->getRetrieveByPkParamsForGetOrCreate() ?>);
 
-      $this->forward404Unless($<?php echo $this->getSingularName() ?>);
+      $this->forward404Unless($this-><?php echo $this->getSingularName() ?>);
+
+      if ($this->save)
+      {
+        $this->update<?php echo $this->getClassName() ?>FromEditRequest();
+      }
     }
 
-    return $<?php echo $this->getSingularName() ?>;
+    return;
   }
 
   protected function processFilters ()
@@ -196,10 +227,24 @@
 <?php if ($this->getParameterValue('list.filters')): ?>
     if ($this->getRequest()->hasParameter('filter'))
     {
+      $filters = $this->getRequestParameter('filters');
+<?php foreach ($this->getColumns('list.filters') as $column): $type = $column->getCreoleType() ?>
+<?php if ($type == CreoleTypes::DATE || $type == CreoleTypes::TIMESTAMP): ?>
+      if (isset($filters['<?php echo $column->getName() ?>']['from']) && $filters['<?php echo $column->getName() ?>']['from'] !== '')
+      {
+        $filters['<?php echo $column->getName() ?>']['from'] = sfI18N::getTimestampForCulture($filters['<?php echo $column->getName() ?>']['from'], $this->getUser()->getCulture());
+      }
+      if (isset($filters['<?php echo $column->getName() ?>']['to']) && $filters['<?php echo $column->getName() ?>']['to'] !== '')
+      {
+        $filters['<?php echo $column->getName() ?>']['to'] = sfI18N::getTimestampForCulture($filters['<?php echo $column->getName() ?>']['to'], $this->getUser()->getCulture());
+      }
+<?php endif; ?>
+<?php endforeach; ?>
+
       $this->getUser()->getAttributeHolder()->removeNamespace('sf_admin/<?php echo $this->getSingularName() ?>/filters');
-      $this->getUser()->getAttributeHolder()->add($this->getRequestParameter('filters'), 'sf_admin/<?php echo $this->getSingularName() ?>/filters');
+      $this->getUser()->getAttributeHolder()->add($filters, 'sf_admin/<?php echo $this->getSingularName() ?>/filters');
     }
-<?php endif ?>
+<?php endif; ?>
   }
 
   protected function processSort ()
@@ -214,18 +259,55 @@
   protected function addFiltersCriteria (&$c)
   {
 <?php if ($this->getParameterValue('list.filters')): ?>
-    $this->filters = $this->getUser()->getAttributeHolder()->getAll('sf_admin/<?php echo $this->getSingularName() ?>/filters');
 <?php foreach ($this->getColumns('list.filters') as $column): $type = $column->getCreoleType() ?>
+<?php if ($type == CreoleTypes::DATE || $type == CreoleTypes::TIMESTAMP): ?>
+    if (isset($this->filters['<?php echo $column->getName() ?>']))
+    {
+      if (isset($this->filters['<?php echo $column->getName() ?>']['from']) && $this->filters['<?php echo $column->getName() ?>']['from'] !== '')
+      {
+<?php if ($type == CreoleTypes::DATE): ?>
+        $criterion = $c->getNewCriterion(<?php echo $this->getPeerClassName() ?>::<?php echo strtoupper($column->getName()) ?>, date('Y-m-d', $this->filters['<?php echo $column->getName() ?>']['from']), Criteria::GREATER_EQUAL);
+<?php else: ?>
+        $criterion = $c->getNewCriterion(<?php echo $this->getPeerClassName() ?>::<?php echo strtoupper($column->getName()) ?>, $this->filters['<?php echo $column->getName() ?>']['from'], Criteria::GREATER_EQUAL);
+<?php endif; ?>
+      }
+      if (isset($this->filters['<?php echo $column->getName() ?>']['to']) && $this->filters['<?php echo $column->getName() ?>']['to'] !== '')
+      {
+        if (isset($criterion))
+        {
+<?php if ($type == CreoleTypes::DATE): ?>
+          $criterion->addAnd($c->getNewCriterion(<?php echo $this->getPeerClassName() ?>::<?php echo strtoupper($column->getName()) ?>, date('Y-m-d', $this->filters['<?php echo $column->getName() ?>']['to']), Criteria::LESS_EQUAL));
+<?php else: ?>
+          $criterion->addAnd($c->getNewCriterion(<?php echo $this->getPeerClassName() ?>::<?php echo strtoupper($column->getName()) ?>, $this->filters['<?php echo $column->getName() ?>']['to'], Criteria::LESS_EQUAL));
+<?php endif; ?>
+        }
+        else
+        {
+<?php if ($type == CreoleTypes::DATE): ?>
+          $criterion = $c->getNewCriterion(<?php echo $this->getPeerClassName() ?>::<?php echo strtoupper($column->getName()) ?>, date('Y-m-d', $this->filters['<?php echo $column->getName() ?>']['to']), Criteria::LESS_EQUAL);
+<?php else: ?>
+          $criterion = $c->getNewCriterion(<?php echo $this->getPeerClassName() ?>::<?php echo strtoupper($column->getName()) ?>, $this->filters['<?php echo $column->getName() ?>']['to'], Criteria::LESS_EQUAL);
+<?php endif; ?>
+        }
+      }
+
+      if (isset($criterion))
+      {
+        $c->add($criterion);
+      }
+    }
+<?php else: ?>
     if (isset($this->filters['<?php echo $column->getName() ?>']) && $this->filters['<?php echo $column->getName() ?>'] !== '')
     {
 <?php if ($type == CreoleTypes::CHAR || $type == CreoleTypes::VARCHAR): ?>
       $c->add(<?php echo $this->getPeerClassName() ?>::<?php echo strtoupper($column->getName()) ?>, strtr($this->filters['<?php echo $column->getName() ?>'], '*', '%'), Criteria::LIKE);
 <?php else: ?>
       $c->add(<?php echo $this->getPeerClassName() ?>::<?php echo strtoupper($column->getName()) ?>, $this->filters['<?php echo $column->getName() ?>']);
-<?php endif ?>
+<?php endif; ?>
     }
-<?php endforeach ?>
-<?php endif ?>
+<?php endif; ?>
+<?php endforeach; ?>
+<?php endif; ?>
   }
 
   protected function addSortCriteria (&$c)
